rpc/dapserver/session.go

```diff
@@ -0,0 +1,1058 @@
+package dapserver
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"path/filepath"
+	"sync"
+
+	"github.com/alecthomas/participle/v2/lexer"
+	dap "github.com/google/go-dap"
+	"github.com/openllb/hlb/codegen"
+	"github.com/openllb/hlb/parser/ast"
+	"github.com/openllb/hlb/pkg/filebuffer"
+)
+
+type Session struct {
+	dbgr codegen.Debugger
+	rw   *bufio.ReadWriter
+
+	cancel context.CancelFunc
+	err    error
+
+	sendQueue chan dap.Message
+	sendWg    sync.WaitGroup
+
+	caps map[Capability]struct{}
+
+	sourcesHandles    *handlesMap
+	variablesHandles  *handlesMap
+	stackFrameHandles *handlesMap
+}
+
+type Capability int
+
+const (
+	VariableTypeCap = iota
+	ProgressReportingCap
+)
+
+func (s *Session) handleRequest(ctx context.Context) error {
+	msg, err := dap.ReadProtocolMessage(s.rw.Reader)
+	if err != nil {
+		return err
+	}
+
+	s.sendWg.Add(1)
```

Comment:
Would there ever be a situation where we get messages faster than we can handle them, and end up spawning too many goroutines here? I'm not sure this is a real concern, as I guess it might imply the sender is really misbehaving, and I'm not sure we need to harden against that.

Reply:
Yeah I'm not sure what DAP specification recommends here. I don't think this is necessary to address in this first PR for DAP support.